.github/actions/test-dotnet/Invoke-SolutionTests.ps1

```diff
@@ -35,21 +37,27 @@ $tests = dotnet sln list |
     }
 
 foreach ($test in $tests) {
-    dotnet test -c Release --no-restore --no-build --nologo --logger "trx;LogFileName=test-results.trx" --verbosity $Verbosity $test 2>&1 >test.out
+    $dotnetTestSwitches = @(
+        '--configuration', 'Release'
+        '--no-restore',
+        '--no-build',
+        '--nologo',
+        '--logger', 'trx;LogFileName=test-results.trx'
+        # This is for xUnit ITestOutputHelper, see https://xunit.net/docs/capturing-output.
+        '--logger', 'console;verbosity=detailed'
+        '--verbosity', $Verbosity
+        [string]::IsNullOrEmpty($Filter) ? '' : "--filter"
+        [string]::IsNullOrEmpty($Filter) ? '' : $Filter
+        $test
+    )
+
+    dotnet test @dotnetTestSwitches
 
     if ($?)
     {
         Write-Output "Test Successful: $test"
         continue
     }
 
-    $needsGrouping = (Select-String "::group::" test.out).Length -eq 0
-
-    if ($needsGrouping) { Write-Output "::group::Test Failed: $test" }
-
-    bash -c "cat test.out | grep -v 'Connection refused \[::ffff:127.0.0.1\]' | grep -v 'ChromeDriver was started successfully'"
-
-    if ($needsGrouping) { Write-Output "::endgroup::" }
-
```

Comment:
Note that despite this missing, there's per-test grouping in the build output, see e.g. [here](https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/actions/runs/3452657800/jobs/5762708404#step:9:80):

![image](https://user-images.githubusercontent.com/1976647/201496395-c469a95a-6687-4480-9884-1f1cfb07af89.png)

Reply:
per-test grouping comes from the UI Testing Toolbox [here](https://github.com/Lombiq/UI-Testing-Toolbox/blob/5d070b36e0cbe99a8bf92aac38f0558b5a876185/Lombiq.Tests.UI/Services/GitHubActionsGroupingTestOutputHelper.cs#L40).
The downside is that now the per-process logs around the tests are still not grouped and when a test succeeds the logs are still printed, polluting the the GHA output.

Reply:
Printing a succeeding test's log as well is desired and one of the goals with these changes. We need to be able to compare successful and failing tests. But right, I'll readd grouping for the processes too.

Reply:
Ah well, there are no nested groups... https://github.com/actions/runner/issues/1477